enhancements/sig-policy/134-standalone-hub-templates/README.md

```diff
@@ -0,0 +1,185 @@
+
+Currently, hub templates are resolved by the Governance Policy Propagator as part of the replicated `Policy` creation.
+In standalone mode, the Governance Policy Propagator and the Governance Policy Framework are not part of the equation.
+Because of this, the simplest and most reliable approach is to add hub template support in the Configuration Policy
+controller.
+
+A way to achieve this is to create an additional addon that is **disabled** by default called
+`governance-standalone-hub-templating`. Enabling this addon will cause the following:
+
+- The addon framework creates a user/subject represented by a certificate on the hub for each managed cluster. There
+  will be no permissions assigned to it.
+- The addon framework will also create a `Secret` in the `open-cluster-management-agent-addon` namespace with the hub
+  user kubeconfig that will be leveraged by the Configuration Policy controller.
+- The Governance Policy Addon controller should configure the `config-policy-controller` to mount the generated
+  kubeconfig when the `governance-standalone-hub-templating` addon is enabled and pass a flag to enable the standalone
+  hub templating mode.
+
+#### Configuration Policy changes
+
+When the Configuration Policy controller encounters a hub template when standalone hub templating mode is disabled, it
+should mark the policy as `NonCompliant` with a message such as
+`The governance-standalone-hub-templating addon must be enabled to resolve hub templates from the cluster`. The
+`ConfigurationPolicy` objects created on the managed cluster through the policy framework will never have hub templates
+since the policy framework marks these policies as `NonCompliant` assuming the hub templates failed to resolve on the
+hub. With this in mind, the user's intentions are safe to assume when a hub template is encountered.
+
+When the Configuration Policy controller has standalone hub templating mode enabled, it should instantiate a separate
+`TemplateResolver` with the `governance-standalone-hub-templating` hub user and resolve hub templates (i.e.
+`{{hub ... hub}}`) prior to managed cluster templates (i.e. `{{ ... }}`). By default, this `TemplateResolver` will have
+no permission on the hub cluster. The hub administrator must grant `list` and `watch` permissions on the objects
+accessed by hub templates. The error messages for lack of permissions are clear due to the work done in the "Expand Hub
+Template Access on Policies" feature.
+
+If the changes stopped here, the case where the managed cluster is always connected to the hub would work well. Short
+disconnects would be okay since objects referenced by the hub templates are cached. Once the Configuration Policy
+controller restarts and thus the cache is cleared, the policy could no longer be evaluated due to hub template
+resolution failing.
+
+The proposed solution is for the Configuration Policy controller to create a `Secret` named `<policy UID>-last-resolved`
```

Comment:
I like this - using the UID avoids a potential issue where the name could otherwise be too long.